src/descriptor.rs

```diff
@@ -83,6 +83,242 @@ pub struct KeyboardReport {
+    KeyboardScrollLock = 0x47,
+    KeyboardPause = 0x48,
+    KeyboardInsert = 0x49,
+    KeyboardHome = 0x4A,
+    KeyboardPageUp = 0x4B,
+    KeyboardDelete = 0x4C,
+    KeyboardEnd = 0x4D,
+    KeyboardPageDown = 0x4E,
+    KeyboardRightArrow = 0x4F,
+    KeyboardLeftArrow = 0x50,
+    KeyboardDownArrow = 0x51,
+    KeyboardUpArrow = 0x52,
+    KeypadNumLock = 0x53,
+    KeypadDivide = 0x54,
+    KeypadMultiply = 0x55,
+    KeypadMinus = 0x56,
+    KeypadPlus = 0x57,
+    KeypadEnter = 0x58,
+    Keypad1End = 0x59,
+    Keypad2DownArrow = 0x5A,
+    Keypad3PageDown = 0x5B,
+    Keypad4LeftArrow = 0x5C,
+    Keypad5 = 0x5D,
+    Keypad6RightArrow = 0x5E,
+    Keypad7Home = 0x5F,
+    Keypad8UpArrow = 0x60,
+    Keypad9PageUp = 0x61,
+    Keypad0Insert = 0x62,
+    KeypadPeriodDelete = 0x63,
+    KeyboardNonUSSlash = 0x64,
+    KeyboardApplication = 0x65,
+    KeyboardPower = 0x66,
+    KeypadEqual = 0x67,
+    KeyboardF13 = 0x68,
+    KeyboardF14 = 0x69,
+    KeyboardF15 = 0x6A,
+    KeyboardF16 = 0x6B,
+    KeyboardF17 = 0x6C,
+    KeyboardF18 = 0x6D,
+    KeyboardF19 = 0x6E,
+    KeyboardF20 = 0x6F,
+    KeyboardF21 = 0x70,
+    KeyboardF22 = 0x71,
+    KeyboardF23 = 0x72,
+    KeyboardF24 = 0x73,
+    KeyboardExecute = 0x74,
+    KeyboardHelp = 0x75,
+    KeyboardMenu = 0x76,
+    KeyboardSelect = 0x77,
+    KeyboardStop = 0x78,
+    KeyboardAgain = 0x79,
+    KeyboardUndo = 0x7A,
+    KeyboardCut = 0x7B,
+    KeyboardCopy = 0x7C,
+    KeyboardPaste = 0x7D,
+    KeyboardFind = 0x7E,
+    KeyboardMute = 0x7F,
+    KeyboardVolumeUp = 0x80,
+    KeyboardVolumeDown = 0x81,
+    KeyboardLockingCapsLock = 0x82,
+    KeyboardLockingNumLock = 0x83,
+    KeyboardLockingScrollLock = 0x84,
+    KeypadComma = 0x85,
+    KeypadEqualAS400 = 0x86,
+    KeyboardInternational1 = 0x87,
+    KeyboardInternational2 = 0x88,
+    KeyboardInternational3 = 0x89,
+    KeyboardInternational4 = 0x8A,
+    KeyboardInternational5 = 0x8B,
+    KeyboardInternational6 = 0x8C,
+    KeyboardInternational7 = 0x8D,
+    KeyboardInternational8 = 0x8E,
+    KeyboardInternational9 = 0x8F,
+    KeyboardLANG1 = 0x90,
+    KeyboardLANG2 = 0x91,
+    KeyboardLANG3 = 0x92,
+    KeyboardLANG4 = 0x93,
+    KeyboardLANG5 = 0x94,
+    KeyboardLANG6 = 0x95,
+    KeyboardLANG7 = 0x96,
+    KeyboardLANG8 = 0x97,
+    KeyboardLANG9 = 0x98,
+    KeyboardAlternateErase = 0x99,
+    KeyboardSysReqAttention = 0x9A,
+    KeyboardCancel = 0x9B,
+    KeyboardClear = 0x9C,
+    KeyboardPrior = 0x9D,
+    KeyboardReturn = 0x9E,
+    KeyboardSeparator = 0x9F,
+    KeyboardOut = 0xA0,
+    KeyboardOper = 0xA1,
+    KeyboardClearAgain = 0xA2,
+    KeyboardCrSelProps = 0xA3,
+    KeyboardExSel = 0xA4,
+    // 0xA5-0xAF: Reserved
+    Keypad00 = 0xB0,
+    Keypad000 = 0xB1,
+    ThousandsSeparator = 0xB2,
+    DecimalSeparator = 0xB3,
+    CurrencyUnit = 0xB4,
+    CurrencySubunit = 0xB5,
+    KeypadOpenPar = 0xB6,
+    KeypadClosePar = 0xB7,
+    KeypadOpenCurlyBrace = 0xB8,
```

Comment:
I prefer the shorter form of { and [

( - parens (or parentheses); par is a bit too short (hard to google for by a non-native speaker)
[ - brackets
{ - braces

See https://en.wikipedia.org/wiki/Bracket